libraries/script-engine/src/qtscript/ScriptEngineQtScript.cpp

```diff
@@ -0,0 +1,1039 @@
+//
+//  ScriptEngineQtScript.cpp
+//  libraries/script-engine/src/qtscript
+//
+//  Created by Brad Hefta-Gaub on 12/14/13.
+//  Copyright 2013 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+#include "ScriptEngineQtScript.h"
+
+#include <chrono>
+#include <thread>
+
+#include <QtCore/QCoreApplication>
+#include <QtCore/QEventLoop>
+#include <QtCore/QFileInfo>
+#include <QtCore/QTimer>
+#include <QtCore/QThread>
+#include <QtCore/QRegularExpression>
+
+#include <QtCore/QFuture>
+#include <QtConcurrent/QtConcurrentRun>
+
+#include <QtWidgets/QMainWindow>
+#include <QtWidgets/QApplication>
+#include <QtWidgets/QMenuBar>
+#include <QtWidgets/QMenu>
+
+#include <QtNetwork/QNetworkRequest>
+#include <QtNetwork/QNetworkReply>
+
+#include <QtScript/QScriptContextInfo>
+#include <QtScript/QScriptValue>
+#include <QtScript/QScriptValueIterator>
+
+#include <QtScriptTools/QScriptEngineDebugger>
+
+#include <shared/LocalFileAccessGate.h>
+#include <shared/QtHelpers.h>
+#include <shared/AbstractLoggerInterface.h>
+
+#include <Profile.h>
+
+#include "../ScriptEngineLogging.h"
+#include "../ScriptProgram.h"
+#include "../ScriptValue.h"
+
+#include "ScriptContextQtWrapper.h"
+#include "ScriptProgramQtWrapper.h"
+#include "ScriptValueQtWrapper.h"
+#include "ScriptContextQtAgent.h"
+
+static const int MAX_DEBUG_VALUE_LENGTH { 80 };
+
+static const QScriptEngine::QObjectWrapOptions DEFAULT_QOBJECT_WRAP_OPTIONS =
+                QScriptEngine::ExcludeDeleteLater | QScriptEngine::ExcludeChildObjects;
+
+Q_DECLARE_METATYPE(ScriptValue);
+
+Q_DECLARE_METATYPE(QScriptEngine::FunctionSignature)
+int qfunctionSignatureMetaID = qRegisterMetaType<QScriptEngine::FunctionSignature>();
+
+int scriptEnginePointerMetaID = qRegisterMetaType<ScriptEngineQtScriptPointer>();
+
+bool ScriptEngineQtScript::IS_THREADSAFE_INVOCATION(const QThread* thread, const QString& method) {
```

Comment:
didn't we have this somewhere else up above?

Reply:
The calls that pass a QThread have been declared as private to the QtScript implementation, where the "current thread" versions are in the "common area".  I'm expecting V8 to be a bit weirder here as I'm hoping to have it only locked to a thread while it's doing something (unsure what will come out of that, possibly nothing).